sdk/src/types/block/output/mod.rs

```diff
@@ -439,55 +440,12 @@ pub trait MinimumOutputAmount: StorageScore {
 pub mod dto {
     use alloc::format;
 
-    use serde::{Deserialize, Serialize, Serializer};
+    use serde::Deserialize;
     use serde_json::Value;
 
     use super::*;
-    pub use super::{
-        account::dto::AccountOutputDto, anchor::dto::AnchorOutputDto, basic::dto::BasicOutputDto,
-        delegation::dto::DelegationOutputDto, foundry::dto::FoundryOutputDto, nft::dto::NftOutputDto,
-    };
-
-    /// Describes all the different output types.
-    #[derive(Clone, Debug, Eq, PartialEq, From)]
-    pub enum OutputDto {
-        Basic(BasicOutputDto),
-        Account(AccountOutputDto),
-        Anchor(AnchorOutputDto),
-        Foundry(FoundryOutputDto),
-        Nft(NftOutputDto),
-        Delegation(DelegationOutputDto),
-    }
-
-    impl From<&Output> for OutputDto {
-        fn from(value: &Output) -> Self {
-            match value {
-                Output::Basic(o) => Self::Basic(o.into()),
-                Output::Account(o) => Self::Account(o.into()),
-                Output::Anchor(o) => Self::Anchor(o.into()),
-                Output::Foundry(o) => Self::Foundry(o.into()),
-                Output::Nft(o) => Self::Nft(o.into()),
-                Output::Delegation(o) => Self::Delegation(o.into()),
-            }
-        }
-    }
 
-    impl TryFrom<OutputDto> for Output {
-        type Error = Error;
-
-        fn try_from(dto: OutputDto) -> Result<Self, Self::Error> {
-            Ok(match dto {
-                OutputDto::Basic(o) => Self::Basic(BasicOutput::try_from(o)?),
-                OutputDto::Account(o) => Self::Account(AccountOutput::try_from(o)?),
-                OutputDto::Anchor(o) => Self::Anchor(AnchorOutput::try_from(o)?),
-                OutputDto::Foundry(o) => Self::Foundry(FoundryOutput::try_from(o)?),
-                OutputDto::Nft(o) => Self::Nft(NftOutput::try_from(o)?),
-                OutputDto::Delegation(o) => Self::Delegation(DelegationOutput::try_from(o)?),
-            })
-        }
-    }
-
-    impl<'de> Deserialize<'de> for OutputDto {
+    impl<'de> Deserialize<'de> for Output {
```

Comment:
We *could* just make this an untagged enum derive now

Reply:
or, rather, just derive both

Reply:
For reference, `Feature` does that and it works fine

Reply:
It does check the next variant but the variant are serialised with their type so even if they have the same fields the type would different